src/NHibernate.Test/NHSpecificTest/NH3818/Fixture.cs

```diff
@@ -40,31 +38,33 @@ public void SelectConditionalValuesTest()
                         .Select(o => new
                         {
                             o.Color,
-                            AliveDays = (int)(DateTime.Now - o.Birthdate).TotalDays,
+                            AliveDays = (now - o.Birthdate).TotalDays,
```

Comment:
Test had start being flaky with `DateTimeType` no more truncating to seconds. The diff followed by `TotalDays` does not count boundaries but a fractional days count. We should not truncate it to int but better instead compare with a tolerance.